tendermint/src/block/header.rs

```diff
@@ -83,11 +85,89 @@ impl lite::Header for Header {
     }
 
     fn next_validators_hash(&self) -> Hash {
-        unimplemented!()
+        self.next_validators_hash
     }
 
     fn hash(&self) -> Hash {
-        unimplemented!()
+        let mut version_enc = vec![];
+        // TODO: if there is an encoding problem this will
+        // panic (as the golang code would):
+        // https://github.com/tendermint/tendermint/blob/134fe2896275bb926b49743c1e25493f6b24cc31/types/block.go#L393
+        // https://github.com/tendermint/tendermint/blob/134fe2896275bb926b49743c1e25493f6b24cc31/types/encoding_helper.go#L9:6
+        // Instead, handle errors gracefully here.
```

Comment:
> Why would there ever be an encoding error and what would it mean to handle it gracefully? 

I can't think of a scenario where there could be and encoding error here (while encoding for  hashing). If there is any case, it still feels wrong to me to panic instead of returning an error and let the caller decide if it is correct to panic (or handle the error in another way). 
If we can not hash the header, this probably means a bug in the encoding library (or another bigger problem) and the caller should probably decide to panic anyways. I just feel it isn't the job of the `hash` method to decide.